lib/improver/tests/utilities/test_TemporalInterpolation.py

```diff
@@ -0,0 +1,183 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for temporal utilities."""
+
+import datetime
+from datetime import timedelta
+import unittest
+import numpy as np
+from cf_units import Unit
+
+from iris.coords import DimCoord
+from iris.exceptions import CoordinateNotFoundError
+from iris.tests import IrisTest
+from iris.cube import Cube
+
+from improver.utilities.temporal import TemporalInterpolation
+
+
+class Test_construct_time_list(IrisTest):
+
+    """Test construction of time lists suitable for iris interpolation using
+    this function."""
+
+    def setUp(self):
+        self.time_0 = datetime.datetime(2017, 11, 1, 3)
+        self.time_1 = datetime.datetime(2017, 11, 1, 9)
+        self.times = []
+        for i in range(4, 9):
+            self.times.append(datetime.datetime(2017, 11, 1, i))
+        self.expected = [('time', time) for time in self.times]
+
+    def test_list_from_interval_in_minutes(self):
+        """Test generating a list between two times using the
+        interval_in_minutes keyword to define the spacing."""
+
+        result = (
+            TemporalInterpolation(interval_in_minutes=60).construct_time_list(
+                self.time_0, self.time_1))
+        self.assertEqual(self.expected, result)
+
+    def test_non_equally_divisible_interval_in_minutes(self):
+        """Test an exception is raised when trying to generate a list of times
+        that would not divide the time range equally."""
+
+        msg = 'interval_in_minutes provided to time_interpolate does not'
+        with self.assertRaisesRegex(ValueError, msg):
+            TemporalInterpolation(interval_in_minutes=61).construct_time_list(
+                self.time_0, self.time_1)
+
+    def test_list_from_existing_list(self):
+        """Test generating an iris interpolation suitable list from an
+        existing list of datetime objects."""
+
+        result = (
+            TemporalInterpolation(times=self.times).construct_time_list(
+                self.time_0, self.time_1))
+        self.assertEqual(self.expected, result)
+
+    def test_time_list_out_of_bounds(self):
+        """Test an exception is raised when trying to generate a list of times
+        using a preexisiting list that inclused times outside the range of the
```

Comment:
I'd probably add a hyphen to 'preexisting'. Dictionary definition allows both, but British dictionaries seem to favour having the hyphen. 

Also, 'inclused' should be 'includes'.

Reply:
✅